Project02Eclipse/src/com/hemebiotech/analytics/WriteSymptomsResultInFile.java

```diff
@@ -0,0 +1,63 @@
+package com.hemebiotech.analytics;
+
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This class contains a method which browse a list of symptoms,
+ * counts the occurrences of each symptom and write the result to a file
+ * @author jonathan.G
+ * @since 1.0
+ */
+public class WriteSymptomsResultInFile implements ISymptomsWriter{
+	
+	private List<String> arraySymptoms;
+	
+	/**
+	 * Constructor
+	 * @param arraySymptoms A list of symptoms
+	 */
+	public WriteSymptomsResultInFile(List<String> arraySymptoms) {
+		this.arraySymptoms = arraySymptoms;
+	}
+	
+	/**
+	 * Extract the differents symptoms and count their occurrences
+	 * Generates a file containing the list of symptoms and the number of times they appear in the file
+	 * @throws IOException 
+	 */
+	public void WriteSymptoms() throws IOException {
+		FileWriter writer = new FileWriter("result.out");
+		ArrayList<String> deduplication = new ArrayList<String>();
```

Comment:
From there, all you code can be simplified in 3 lines 😇
Tips -> look at the `Collection` java documentation. There is a `frequency ` method, which could help you a lot.

https://docs.oracle.com/javase/7/docs/api/java/util/Collections.html#frequency(java.util.Collection,%20java.lang.Object)

Reply:
It's done but there are a little more than 3 lines.
you will tell tomorow me if it's what you expected :)